zeebe/journal/src/main/java/io/camunda/zeebe/journal/file/SegmentedJournal.java

```diff
@@ -187,6 +190,13 @@ public boolean isOpen() {
     return open;
   }
 
+  @Override
+  public SortedMap<Long, Path> getTailSegments(final long index) {
+    // lazy view on the map
```

Comment:
:question: I've never used this, but if I understand correctly, this means the transform function is applied lazily, i.e. only when the element is accessed, correct?

Considering this is a view, and the underlying map is mutable, should we not eat the cost of generating an immutable copy (anyway the map should not be very large)? I think it would make it a lot easier to reason about, i.e. the tail segments at calling time.

Reply:
Yes, even though the underlying map is concurrent, it's better to have an immutable copy at a specific point in time.